File: src/main/java/com/component/authserver/component/ButtonAnchor.java
package com.component.authserver.component;

import com.vaadin.flow.component.AttachEvent;
import com.vaadin.flow.component.Component;
import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.UI;
import com.vaadin.flow.component.Unit;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.button.ButtonVariant;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.html.Anchor;
import com.vaadin.flow.component.html.Image;
import com.vaadin.flow.component.icon.Icon;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.component.template.Id;
import com.vaadin.flow.server.StreamResource;
import org.springframework.util.StringUtils;

@Tag("button-anchor")
@JsModule("./components/button-anchor.js")
public class ButtonAnchor extends LitTemplate {

    @Id("button")
    private Button button;
    @Id("img")
    private Image img;
    @Id("anchor")
    private Anchor anchor;

    public ButtonAnchor(String src, String filename, String text, String href, String backgroundColor, String textColor){
        StreamResource imageResource = new StreamResource(filename,
                () -> getClass().getResourceAsStream(src));
        Image icon = new Image(imageResource,"");
        button.setText(text);
        button.setIcon(icon);
        button.getStyle().set("background-color", backgroundColor);
        button.getStyle().set("color", textColor);
        getElement().setProperty("src", src);
        getElement().setProperty("name", text);
        getElement().setProperty("href", href);
    }

}

File: frontend/component/button-anchor.js

import { html, LitElement } from 'lit';
import '@vaadin/button';

class ButtonAnchor extends LitElement {

    static get properties() {
        return {
            scr: { type: String },
            name: { type: String },
            href: { type: String }
        }
    }
    render() {
        return html`
                    <a router-ignore id="anchor" href="${this.href}" style="text-decoration: none; color: black;background-color:transparent !important;">
                        <vaadin-button id="button" style="margin: 0; padding: 0">
                            <img src="${this.src}" id="img">${this.name}
                        </vaadin-button>
                    </a>
                
        `;
    }

}

customElements.define('button-anchor', ButtonAnchor);